networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/rpc/RpcHandler.java

```diff
@@ -169,11 +194,25 @@ public void setRequestHandler(RpcRequestHandler rpcRequestHandler) {
         throw new IllegalStateException("Attempt to set an already set data handler");
       }
       this.rpcRequestHandler = rpcRequestHandler;
-      activeFuture.thenAccept(__ -> rpcRequestHandler.onActivation(rpcStream)).reportExceptions();
-      while (!bufferedData.isEmpty()) {
-        ByteBuf currentBuffer = bufferedData.remove(0);
-        this.rpcRequestHandler.onData(nodeId, rpcStream, currentBuffer);
-      }
+
+      activeFuture
+          .thenCompose(
+              __ ->
+                  asyncRunner.runAsync(
```

Comment:
(discussion) Here, we're processing the input asynchronously using the standard `DelayedExecutorAsyncRunner`.  Wondering if we should run these tasks in a dedicated thread pool?

Reply:
Yeah I've been generally trying to ensure we use a single async runner that's injected rather than creating them all over the place so we can start to make decisions about how many threads it can use etc.

It probably makes sense to have networking use it's own thread pool.  If we're blocking we may need to make the limit on number of threads quite high, but it would definitely be good to know how many threads are being used by networking and be able to limit them independently.

Reply:
I think this might require a bit of reorganization and this PR is already pretty big, so I think I'll log a ticket to follow-up on this if that sounds fine to you.

Reply:
Yep agreed.